src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -445,6 +445,14 @@ private string GetDebuggerDisplay()
         /// </summary>
         internal abstract Diagnostic WithIsSuppressed(bool isSuppressed);
 
+        /// <summary>
+        /// Create a new instance of this diagnostic with the given programmatic suppression info.
+        /// </summary>
+        internal Diagnostic WithProgrammaticSuppression(ProgrammaticSuppressionInfo programmaticSuppressionInfo)
```

Comment:
>WithProgrammaticSuppression [](start = 28, length = 27)

Is it valid to call this method on ```DiagnosticWithProgrammaticSuppression```?

Reply:
It is not expected, I will add an assert.